### PR TITLE
Support for add relu functional module

### PR DESCRIPTION
**Pull request by raghuramank100**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26612 Support for add relu functional module**

Add support for add relu functional module, this allows for fusion of add and relu quantized operations

Differential Revision: [D17518268](https://our.internmc.facebook.com/intern/diff/D17518268/)

### CODE REVIEW

File: test/common_quantization.py [hunk 1]
```diff
@@ -451,22 +451,27 @@ def __init__(self, quantized=False):
...
     def forward(self, x):
         y = self.mycat.cat([x, x, x])
         z = self.myadd.add(y, y)
-        return z
+        w = self.myadd_relu.add_relu(z, z)
+        return w
 
     @classmethod
     def from_float(cls, mod):
         new_mod = cls(quantized=True)
```
Comment (jerryzh168) [inline test/common_quantization.py]: why do you need to do this? isn't this done by the flow?

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
I think there is some problem with original test

**Review by dskhudia — APPROVED** [review]
LGTM!

File: test/common_quantization.py (hunk 1 above)
Comment (raghuramank100) [reply test/common_quantization.py]: Rewrote the tests to match how we test other modules

**Review by dzhulgakov — APPROVED** [review]

**Review by jerryzh168 — APPROVED** [review]
Looks good

File: test/test_quantization.py [hunk 2]
```diff
@@ -670,31 +670,35 @@ def test_single_layer(self):
...
-        self.qx = torch.quantize_per_tensor(self.x.to(torch.float), scale=1.0,
-                                            zero_point=0, dtype=torch.qint32)
+class FunctionalModuleTest(QuantizationTestCase):
+    # Histogram Observers are slow, so have no-deadline to ensure test doesn't time out
+    @no_deadline
+    @given(train_mode=st.booleans())
+    def test_functional_module(self, train_mode):
+        model = ModelWithFunctionals()
+        xq = torch.quantize_per_tensor(torch.rand(10, 1, dtype=torch.float), 0.01, 30, torch.quint8)
+        self.checkScriptable(model, [(xq.dequantize(), xq.dequantize())], check_save_load=True)
```
Comment (zafartahirov) [inline test/test_quantization.py]: nit: factor out the `xq.dequantize()` -- it is used in multiple places.

**Review by z-a-f — APPROVED** [review]
LGTM

**Pull request closed without merging**